File: searching/nestedRangesCheck.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <limits>
using namespace std;

vector<int> gai() {
    int l, r;
    cin >> l >> r;
    return {l, r};
}

void test() {
    int n;
    cin >> n;
    vector<pair<pair<int, int>, int>> intervals;
    vector<int> contained(n, 0); 
    vector<int> contains(n, 0);
    for (int i = 0; i < n; ++i) {
        auto interval = gai();
        intervals.push_back({{interval[0], interval[1]}, i});
    }


    sort(intervals.begin(), intervals.end(), [](const auto &a, const auto &b) {
        return a.first.first < b.first.first || 
               (a.first.first == b.first.first && a.first.second > b.first.second);
    });

    int maxR = numeric_limits<int>::min();
    for (const auto &interval : intervals) {
        int l = interval.first.first, r = interval.first.second, i = interval.second;
        if (r <= maxR) {
            contained[i] = 1;
        }
        maxR = max(maxR, r);
    }

    int minR = numeric_limits<int>::max();
    for (int j = n - 1; j >= 0; --j) {
        int l = intervals[j].first.first, r = intervals[j].first.second, i = intervals[j].second;
        if (r >= minR) {
            contains[i] = 1;
        }
        minR = min(minR, r);
    }

    for (const auto &val : contains) {
        cout << val << " ";
    }
    cout << "\n";
    for (const auto &val : contained) {
        cout << val << " ";
    }
    cout << "\n";
}

int main() {
    test();
    return 0;
}

File: searching/nestedRangesCheck.py
import io,os, sys
from itertools import *
import math
import bisect
 
input = sys.stdin.readline
print = lambda x: sys.stdout.write(str(x))
 
alphs = "abcdefghijklmnopqrstuvwxyz"
digs = "0123456789"
 
INT_MIN = -1e10
INT_MAX = 1e10
MODVALUE = 10 ** 9 + 7
 
def gai():
    return list(map(int, input().strip().split()))
 
def test():
    n = int(input())
    intervals = []
    for i in range(n):
        i = gai()
        intervals.append(i)
    contained = [0] * n
    contains = [0] * n

    intervals = [(l, r, i) for i, (l, r) in enumerate(intervals)]

    intervals.sort(key=lambda x: (x[0], -x[1]))
    maxR = -float('inf')
    for l, r, i in intervals:
        if r <= maxR:
            contained[i] = 1
        maxR = max(maxR, r)

    minR = float('inf')
    for j in range(n - 1, -1, -1):
        l, r, i = intervals[j]
        if r >= minR:
            contains[i] = 1
        minR = min(minR, r)
    for i in contains:
        print(i)
        print(" ")
    print("\n")
    for i in contained:
        print(i)
        print(" ")
    return
    
 
test()